cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -337,10 +345,9 @@ private predicate nodeCand1(Node node, boolean stored, Configuration config) {
     )
     or
     // store
-    exists(Node mid, Content f |
-      store(node, f, mid) and
+    exists(Content f |
+      nodeCand1Store(f, node, config) and
       readCand1(f, unbind(config)) and
```

Comment:
Similarly here, the `unbind` shouldn't be necessary, as the optimiser ought to then do a direct multi-column join on the two recursive calls.